### PR TITLE
Add export_w_elements to Q2d

### PR DESCRIPTION
**Pull request by isaacansys**
Add a method to export all available variation W-elements from a 2D
Extractor design.

### CODE REVIEW

File: _unittest/test_36_Q2D_PostProcessing.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class TestClass(BasisTest, object):
+    def setup_class(self):
+        BasisTest.my_setup(self)
+
+    def teardown_class(self):
+        BasisTest.my_teardown(self)
+
+    def test_01_export_w_elements_from_sweep(self):
+        test_project = self.local_scratch.copyfile(os.path.join(local_path, "example_models", "q2d_solved_sweep.aedtz"))
```
Comment (isaacansys) [inline _unittest/test_36_Q2D_PostProcessing.py:18]: Not sure the best way to work with multiple projects in a single test class.

File: pyaedt/q3d.py [hunk 2]
```diff
@@ -1017,6 +1017,129 @@ def auto_assign_conductors(self):
...
+        if analyze:
+            self.analyze_all()
+        setups = self.oanalysis.GetSetups()
+
+        for s in setups:
+            sweeps = self.oanalysis.GetSweeps(s)
+            if len(sweeps) == 0:
+                sweeps = ["LastAdaptive"]
+            else:
+                pass
```
Comment (akaszynski) [inline pyaedt/q3d.py:1052]: This else doesn't seem necessary.

File: pyaedt/q3d.py [hunk 3]
```diff
@@ -1017,6 +1017,129 @@ def auto_assign_conductors(self):
...
+                                "RiseTime:=",
+                                "1e-09s",
+                            ],
+                            subckt_name,
+                            "WElement",
+                            0,
+                        )
+                        exported_files.append(export_path)
+                        self.logger.info("Exported W-element: %s", export_path)
+                    except:
```
Comment (akaszynski) [inline pyaedt/q3d.py:1095]: Doesn't look like we'll be able to hit this with unit testing.

```suggestion
                    except:  # pragma: no cover
```

File: pyaedt/q3d.py [hunk 4]
```diff
@@ -1017,6 +1017,129 @@ def auto_assign_conductors(self):
...
+                                    "RiseTime:=",
+                                    "1e-09s",
+                                ],
+                                subckt_name,
+                                "WElement",
+                                0,
+                            )
+                            exported_files.append(export_path)
+                            self.logger.info("Exported W-element: %s", export_path)
+                        except:
```
Comment (akaszynski) [inline pyaedt/q3d.py:1139]: Same here.

```suggestion
                        except:  # pragma: no cover
```

File: pyaedt/q3d.py [hunk 5]
```diff
@@ -1017,6 +1017,129 @@ def auto_assign_conductors(self):
             self.logger.info("No new nets identified")
         return True
 
+    @pyaedt_function_handler()
+    def export_w_elements(self, analyze=False, export_folder=None):
+        """Export all available W-elements to files.
+
+        Parameters
+        ----------
+        analyze : bool
```
Comment (MaxJPRey) [inline pyaedt/q3d.py:1026]:
```suggestion
        analyze : bool, optional
```

File: pyaedt/q3d.py [hunk 6]
```diff
@@ -1017,6 +1017,129 @@ def auto_assign_conductors(self):
...
+            export_folder = self.working_directory
+        if not os.path.exists(export_folder):
+            os.makedirs(export_folder)
+        if analyze:
+            self.analyze_all()
+        setups = self.oanalysis.GetSetups()
+
+        for s in setups:
+            sweeps = self.oanalysis.GetSweeps(s)
+            if len(sweeps) == 0:
```
Comment (MaxJPRey) [inline pyaedt/q3d.py:1049]:
```suggestion
            if not sweeps:
```

**Comment by isaacansys** [comment]
@MaxJPRey @akaszynski I've incorporated your feedback and added a new test.  Thank you!

**Review by MaxJPRey — APPROVED** [review]

**Pull request merged**